### PR TITLE
Chore/psalm

### PR DESCRIPTION
**Pull request by RobjS**
This PR adds Psalm and modifies the codebase to make it Psalm-compliant in PHP 7.4-8.2 inclusive.

I've modified the CI to explicitly run PHP in each of the PHP versions we run in that environment, because by default Psalm runs in the PHP version inferred from the composer platform (currently 7.4). That would mean, even though we run CI in 4 different PHP versions, Psalm would be running against 7.4 in all 4 pipelines. Now, it will run once against 7.4 (as part of the standard `script/test` run), but then also run against the PHP version set for that pipeline.

Resolves https://github.com/dxw/whippet/issues/222

- [ ] Includes tests (if new features are introduced) - N/A
- [x] Commit message includes link to the ticket/issue
- [x] Changelog updated
- [ ] PR open against [dxw's Homebrew Tap](https://github.com/dxw/homebrew-tap/) to point the Whippet formula to the new version number  - N/A

### CODE REVIEW

File: src/Git/Git.php [hunk 1]
```diff
@@ -81,7 +81,7 @@ public function clone_no_checkout($repository)
 			return false;
 		}
 
-		$this->run_command("mv {$tmpdir}/.git {$this->repo_path}");
+		$this->run_command(['mv', $tmpdir . '/.git', $this->repo_path]);
```
Comment (snim2) [inline src/Git/Git.php:84]: Non-blocking comment: there's a typo in the commit message here `s/no/not/g`

**Review by snim2 — COMMENTED** [review]
This looks good. The return type thing is quite unfortunate though, and I think we should probably have a PHP8.x issue that lists things like this that need tidying up.

File: src/Git/Git.php (hunk 1 above)
Comment (RobjS) [reply src/Git/Git.php:84]: Good spot! Fixed in a rebase I'll push post-approval.

**Comment by RobjS** [comment]
@snim2 This is ready for another look now.

**Review by snim2 — APPROVED** [review]

**Pull request merged**